src/Marten/Internal/Sessions/EventTracingConnectionLifetime.cs

```diff
@@ -0,0 +1,286 @@
+    {
+        return _innerConnectionLifetime.DisposeAsync();
+    }
+
+    public void Dispose()
+    {
+        _innerConnectionLifetime.Dispose();
+    }
+
+    public IMartenSessionLogger Logger { get; set; }
+    public int CommandTimeout { get; }
+    public int Execute(NpgsqlCommand cmd)
+    {
+        if (_openTelemetryOptions.TrackConnectionEvents)
+        {
+            _databaseActivity.AddEvent(new ActivityEvent(MartenCommandExecutionStarted));
+        }
+
+        try
+        {
+            return _innerConnectionLifetime.Execute(cmd);
+        }
+        catch (Exception e)
+        {
+            if (_openTelemetryOptions.TrackConnectionEvents)
+            {
+                _databaseActivity.AddEvent(new ActivityEvent(MartenCommandExecutionFailed, DateTimeOffset.UtcNow, new ActivityTagsCollection(new KeyValuePair<string, object>[] {new KeyValuePair<string, object>("ExceptionType", e.GetType()) })));
+            }
+
+            throw;
+        }
+    }
+
+    public Task<int> ExecuteAsync(NpgsqlCommand command, CancellationToken token = new CancellationToken())
+    {
+        if (_openTelemetryOptions.TrackConnectionEvents)
+        {
+            _databaseActivity.AddEvent(new ActivityEvent("Database command execution started"));
+        }
+
+        try
+        {
+            return _innerConnectionLifetime.ExecuteAsync(command, token);
+        }
+        catch (Exception e)
+        {
+            if (_openTelemetryOptions.TrackConnectionEvents)
+            {
+                _databaseActivity.AddEvent(new ActivityEvent(MartenCommandExecutionFailed, DateTimeOffset.UtcNow,
+                    new ActivityTagsCollection(new[]
+                    {
+                        new KeyValuePair<string, object>("ExceptionType", e.GetType())
+                    })));
+            }
+
+            throw;
+        }
+    }
+
+    public DbDataReader ExecuteReader(NpgsqlCommand command)
+    {
+        if (_openTelemetryOptions.TrackConnectionEvents)
+        {
+            _databaseActivity.AddEvent(new ActivityEvent("Database command execution started"));
+        }
+
+        try
+        {
+            return _innerConnectionLifetime.ExecuteReader(command);
+        }
+        catch (Exception e)
+        {
+            if (_openTelemetryOptions.TrackConnectionEvents)
+            {
+                _databaseActivity.AddEvent(new ActivityEvent(MartenCommandExecutionFailed, DateTimeOffset.UtcNow,
+                    new ActivityTagsCollection(new[]
+                    {
+                        new KeyValuePair<string, object>("ExceptionType", e.GetType())
+                    })));
+            }
+
+            throw;
+        }
+    }
+
+    public Task<DbDataReader> ExecuteReaderAsync(NpgsqlCommand command, CancellationToken token = default)
+    {
+        if (_openTelemetryOptions.TrackConnectionEvents)
+        {
+            _databaseActivity.AddEvent(new ActivityEvent("Database command execution started"));
```

Comment:
Event name should be a constant too, and have a name following OTel semantics (see first comment).